src/Analyzers/Core/Analyzers/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -50,91 +66,74 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
 
-                var objectType = startContext.Compilation.GetSpecialType(SpecialType.System_Object);
-                var referenceEqualsMethodOpt = objectType?.GetMembers(nameof(ReferenceEquals))
+                var objectType = context.Compilation.GetSpecialType(SpecialType.System_Object);
+                var referenceEqualsMethod = objectType?.GetMembers(nameof(ReferenceEquals))
                                                           .OfType<IMethodSymbol>()
                                                           .FirstOrDefault(m => m.DeclaredAccessibility == Accessibility.Public &&
                                                                                m.Parameters.Length == 2);
 
                 var syntaxKinds = GetSyntaxFacts().SyntaxKinds;
-                startContext.RegisterSyntaxNodeAction(
-                    c => AnalyzeSyntax(c, expressionTypeOpt, referenceEqualsMethodOpt),
+                context.RegisterSyntaxNodeAction(
+                    context => AnalyzeTernaryConditionalExpression(context, expressionType, referenceEqualsMethod),
                     syntaxKinds.Convert<TSyntaxKind>(syntaxKinds.TernaryConditionalExpression));
+                context.RegisterSyntaxNodeAction(
+                    context => AnalyzeIfStatement(context, referenceEqualsMethod),
+                    IfStatementSyntaxKind);
             });
-
         }
 
-        private void AnalyzeSyntax(
+        private void AnalyzeTernaryConditionalExpression(
             SyntaxNodeAnalysisContext context,
-            INamedTypeSymbol? expressionTypeOpt,
-            IMethodSymbol? referenceEqualsMethodOpt)
+            INamedTypeSymbol? expressionType,
+            IMethodSymbol? referenceEqualsMethod)
         {
             var conditionalExpression = (TConditionalExpressionSyntax)context.Node;
 
             var option = context.GetAnalyzerOptions().PreferNullPropagation;
             if (!option.Value)
-            {
                 return;
-            }
 
             var syntaxFacts = GetSyntaxFacts();
             syntaxFacts.GetPartsOfConditionalExpression(
-                conditionalExpression, out var conditionNode, out var whenTrueNode, out var whenFalseNode);
+                conditionalExpression, out var condition, out var whenTrue, out var whenFalse);
 
-            conditionNode = syntaxFacts.WalkDownParentheses(conditionNode);
-            whenTrueNode = syntaxFacts.WalkDownParentheses(whenTrueNode);
-            whenFalseNode = syntaxFacts.WalkDownParentheses(whenFalseNode);
+            var conditionNode = (TExpressionSyntax)condition;
 
-            var conditionIsNegated = false;
-            if (syntaxFacts.IsLogicalNotExpression(conditionNode))
-            {
-                conditionIsNegated = true;
-                conditionNode = syntaxFacts.WalkDownParentheses(
-                    syntaxFacts.GetOperandOfPrefixUnaryExpression(conditionNode));
-            }
```

Comment:
this removed check, and the removed check below both moved into TryAnalyzeCondition